openapi-generator/test/generator_arguments_test.dart

```diff
@@ -0,0 +1,205 @@
+        cachePath: 'test',
+        outputDirectory: 'path',
+        templateDirectory: 'template',
+        runSourceGen: false,
+        wrapper: Wrapper.fvm,
+        generator: Generator.dioAlt,
+        skipValidation: true,
+        fetchDependencies: false,
+        inputSpecFile: 'test.yaml',
+        importMapping: {'key': 'value'},
+        typeMapping: {'package': 'type'},
+        reservedWordsMapping: {'const': 'final'},
+        inlineSchemaNameMapping: {'L': 'R'},
+        // TODO: How do I test this?
+        // additionalProperties: {'allowNull': false as DartObject},
+        // inlineSchemaOptions: {'allowNull': false as DartObject},
+      );
+      test('alwaysRun', () => expect(args.alwaysRun, isTrue));
+      test('useNextGen', () => expect(args.useNextGen, isTrue));
+      test('cachePath', () => expect(args.cachePath, 'test'));
+      test('outputDirectory', () => expect(args.outputDirectory, 'path'));
+      test('runSourceGen', () => expect(args.runSourceGen, isFalse));
+      test('shouldFetchDependencies',
+          () => expect(args.shouldFetchDependencies, isFalse));
+      test('skipValidation', () => expect(args.skipValidation, isTrue));
+      test('inputFile', () => expect(args.inputFile, 'test.yaml'));
+      test('templateDirectory',
+          () => expect(args.templateDirectory, 'template'));
+      test('generator', () => expect(args.generator, Generator.dioAlt));
+      test('wrapper', () => expect(args.wrapper, Wrapper.fvm));
+      test('importMappings',
+          () => expect(args.importMappings, {'key': 'value'}));
+      test(
+          'typeMappings', () => expect(args.typeMappings, {'package': 'type'}));
+      test('reservedWordsMappings',
+          () => expect(args.reservedWordsMappings, {'const': 'final'}));
+      test('inlineSchemaNameMappings',
+          () => expect(args.inlineSchemaNameMappings, {'L': 'R'}));
+      test(
+          'additionalProperties',
+          () => expect(
+              args.additionalProperties, {'allowNull': false as DartObject}),
+          skip: true);
+      test(
+          'inlineSchemaOptions',
+          () => expect(
+              args.inlineSchemaOptions, {'allowNull': false as DartObject}),
+          skip: true);
+
+      test('generatorName', () => expect(args.generatorName, 'dart2-api'));
+      test('shouldGenerateSources',
+          () => expect(args.shouldGenerateSources, isTrue));
+      test(
+          'jarArgs',
+          () => expect(args.jarArgs, [
+                'generate',
+                '-o path',
+                '-i test.yaml',
+                '-t template',
+                '-g dart2-api',
+                '--skip-validate-spec',
+                '--reserved-words-mappings=const=final',
+                '--inline-schema-name-mappings=L=R',
+                '--import-mappings=key=value',
+                '--type-mappings=package=type'
+              ]));
+    });
+    test('uses config', () async {
```

Comment:
I'm not sure how best to test this. I need to build a constantreader

Reply:
Test the jarargs? Just confirming it generates the right command should be enough

Reply:
This is mostly to ensure that the expected values are injected correctly.